### PR TITLE
feat(toolkit): recursive repository cloning

### PR DESCRIPTION
**Pull request by badayvedat**

### CODE REVIEW

File: projects/fal/src/fal/toolkit/utils/download_utils.py [hunk 1]
```diff
@@ -387,6 +388,18 @@ def clone_repository(
...
+    try:
+        return subprocess.check_output(
+            ["git", "rev-parse", "HEAD"],
+            cwd=repo_path,
+            text=True,
+            stderr=subprocess.STDOUT,
+        ).strip()
+    except subprocess.CalledProcessError as error:
+        if "not a git repository" in error.output:
+            return ""
```
Comment (mederka) [inline projects/fal/src/fal/toolkit/utils/download_utils.py:400]: Shouldn't we let the user know somehow?

**Review by mederka — COMMENTED** [review]
ATL: what is the purpose of recursive cloning here?

**Comment by badayvedat** [comment]
some model repositories contain folders that reference some other repositories, which requires recursive cloning. clone_repository fails to address this issue at the moment

**Review by mederka — APPROVED** [review]
Thanks for the clarification 👍

**Pull request merged**